dev_guide/builds.adoc

```diff
@@ -978,6 +978,59 @@ containing an inline Dockerfile that potentially replaces one within the
 contents of the binary archive.
 ====
 
+[[image-source]]
+
+== Image Source
+
+Additional files can be provided to the build process via images.  Input
+images are referenced in the same way the `From` and `To` image targets
+are defined.  This means both docker images and image stream
+tags can be referenced.  In conjunction with the image, you must provide one or 
+more path pairs to indicate the path of the files/directories to copy out of the
+image and the destination to place them in the build context.
+
+The source path can be any location within the image specified.  The destination
+must be a relative directory path.  At build time, the image will be loaded
+and the indicated files/directories will be copied into the context directory
+of the build process.  This is the same directory into which the source repository
+content (if any) is cloned.
+
+Image inputs are specified in the `*source*` definition of the `*BuildConfig*`:
+
+====
+
+----
+    source:
+      git:
+        uri: https://github.com/openshift/ruby-hello-world.git
+      images: <1>
+      - from: <2>
+          kind: ImageStreamTag
+          name: myinputimage:latest
+          namespace: mynamespace
+        paths: <3>
+        - destinationDir: injected/dir <4>
+          sourcePath: /usr/lib/somefile.jar <5>
+      - from:
+          kind: ImageStreamTag
+          name: myotherinputimage:latest
+          namespace: myothernamespace
+        pullSecret: mysecret <6>
+        paths:
+        - destinationDir: injected/dir
+          sourcePath: /usr/lib/somefile.jar
+----
+
+<1> An array of one or more input images and files.
+<2> A reference to the image containing the files to be copied.
+<3> An array of source/destination paths.
+<4> The build process (assemble script or Dockerfile) can reference the file at this relative path.
```

Comment:
A description of what it is may be more straightforward:
'The directory relative to the build root where the build process can access the file'

One thing to note that I just realized is that the location is relative to the place where we extract the source and not the directory that we use for the build. For example, you may have a git repo with a  context dir of 'foo/bar' and an image source with destinationDir of 'foo/baz'. The files from the image will end up copied outside of the directory we're using for building.